src/__tests__/AppLayout.test.tsx

```diff
@@ -2,9 +2,16 @@ import { render } from '@testing-library/react';
 import { AppLayout } from '../AppLayout';
 import { MemoryRouter } from 'react-router-dom';
 
+jest.mock('../services/keycloakService');
+
 window.pnc = {
   config: {
     userGuideUrl: 'https://localhost:3000/',
+    keycloak: {
+      url: 'https://localhost:3000/',
+      realm: 'test',
+      clientId: 'test',
+    },
   },
 };
 
```

Comment:
Since now we have several interactions with the front end about the login/user operation, you can implement some unit tests to mock the login and log out process, for example, to verify whether the user name is shown properly as the response from keycloak passes, or whether logout process can proceed successfully and user can log in again with the new user identity. You can use the mock keycloak service you added.